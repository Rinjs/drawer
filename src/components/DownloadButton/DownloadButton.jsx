import React from "react";

export const DownloadButton = ({ href }) => {
  const buttonStyle = href
    ? "mb-3 display-4 btn btn-light stretched-link"
    : "mb-3 alert alert-light";
  const Component = href ? "a" : "p";
  return (
    <Component
      href={href}
      download={href ? "output" : null}
      className={buttonStyle}
    >
      {href ? "Download canvas" : "Upload your .txt file with commands"}
    </Component>
  );
};
