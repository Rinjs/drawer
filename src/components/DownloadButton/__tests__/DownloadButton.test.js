import React from "react";
import renderer from "react-test-renderer";
import { DownloadButton } from "../DownloadButton";

test("renders without href", () => {
  const component = renderer.create(<DownloadButton href={false} />);
  expect(component).toMatchSnapshot();
});

test("renders with href", () => {
  const component = renderer.create(<DownloadButton href={"something"} />);
  expect(component).toMatchSnapshot();
});
