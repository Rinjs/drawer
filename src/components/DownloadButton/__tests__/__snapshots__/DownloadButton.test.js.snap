// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with href 1`] = `
<a
  className="mb-3 display-4 btn btn-light"
  download="output"
  href="something"
  style={
    Object {
      "fontSize": "something",
      "textAlign": "center",
    }
  }
>
  Download canvas
</a>
`;

exports[`renders without crashing 1`] = `
<p
  className="mb-3 alert alert-light"
  download={null}
  href={false}
  style={
    Object {
      "fontSize": "87.5%",
      "textAlign": "center",
    }
  }
>
  Upload your .txt file with commands
</p>
`;

exports[`renders without href 1`] = `
<p
  className="mb-3 alert alert-light"
  download={null}
  href={false}
  style={
    Object {
      "fontSize": "87.5%",
      "textAlign": "center",
    }
  }
>
  Upload your .txt file with commands
</p>
`;
